src/cli/add.rs

```diff
@@ -498,22 +510,73 @@ fn update_conda_specs_from_lock_file(
+    let (min_version, max_version) = versions.clone().into_iter().minmax().into_option()?;
     let lower_bound = min_version.clone();
     let upper_bound = max_version
         .pop_segments(1)
         .unwrap_or_else(|| max_version.clone())
         .bump(VersionBumpType::Last)
         .ok()?;
-    Some(VersionSpec::Group(
-        LogicalOperator::And,
-        vec![
-            VersionSpec::Range(RangeOperator::GreaterEquals, lower_bound),
-            VersionSpec::Range(RangeOperator::Less, upper_bound),
-        ],
-    ))
+    let constraint = match pinning_strategy {
+        PinningStrategy::ExactVersion => VersionSpec::Group(
+            LogicalOperator::Or,
+            versions
+                .into_iter()
+                .dedup()
+                .map(|v| VersionSpec::Exact(EqualityOperator::Equals, v.clone()))
+                .collect(),
+        ),
+        PinningStrategy::Major => {
+            let upper_bound = max_version
+                .pop_segments(2)
+                .unwrap_or_else(|| upper_bound.clone())
+                .bump(VersionBumpType::Major)
+                .ok()?;
+            VersionSpec::Group(
+                LogicalOperator::And,
+                vec![
+                    VersionSpec::Range(RangeOperator::GreaterEquals, lower_bound),
+                    VersionSpec::Range(RangeOperator::Less, upper_bound),
+                ],
+            )
+        }
+        PinningStrategy::Minor => {
+            let upper_bound = max_version
+                .pop_segments(1)
+                .unwrap_or_else(|| upper_bound.clone())
+                .bump(VersionBumpType::Minor)
+                .ok()?;
+            VersionSpec::Group(
+                LogicalOperator::And,
+                vec![
+                    VersionSpec::Range(RangeOperator::GreaterEquals, lower_bound),
+                    VersionSpec::Range(RangeOperator::Less, upper_bound),
+                ],
+            )
+        }
+        PinningStrategy::LatestUp => VersionSpec::Range(RangeOperator::GreaterEquals, lower_bound),
+        PinningStrategy::NoPin => VersionSpec::Any,
+        PinningStrategy::Semver => {
+            // For v0 packages, we pin to minor instead of major. As discussed in https://github.com/prefix-dev/pixi/issues/1562
+            let is_v0 = max_version.starts_with(&Version::from_str("0").expect("0 is version"));
+            let upper_bound = max_version
+                .pop_segments(if is_v0 { 1 } else { 2 })
+                .unwrap_or_else(|| upper_bound.clone())
+                .bump(VersionBumpType::Last)
+                .ok()?;
+            VersionSpec::Group(
+                LogicalOperator::And,
+                vec![
+                    VersionSpec::Range(RangeOperator::GreaterEquals, lower_bound),
+                    VersionSpec::Range(RangeOperator::Less, upper_bound),
+                ],
+            )
+        }
```

Comment:
This is not technically correct. It should pin the first left-most non-zero. So `0.0.1` should pin the patch version.

Reply:
Changed the behavior, check the tests for the results